<launch>
  <arg name="FC_MODE" default="true"/>
  <arg name="OCS_MODE" default="false" />

  <arg name="CONTROLLER_DEV" default="/dev/input/js0" />
  <arg name="OCS_NS" default="ocs" />
  <arg name="LAUNCH_HANDLE_DETECTOR" default="$(arg FC_MODE)" />
  <arg name="LAUNCH_UI" default="$(arg OCS_MODE)" />
  <arg name="LAUNCH_STEERING" default="false" />
  <arg name="LAUNCH_PS3JOY" default="$(arg OCS_MODE)" />
  <arg name="LAUNCH_EUS" default="$(arg FC_MODE)" />
  <arg name="LAUNCH_RVIZ" default="$(arg OCS_MODE)" />
  <arg name="LAUNCH_CAR_CENTER" default="$(arg FC_MODE)" />
  <arg name="LAUNCH_RECOGNITION" default="$(arg FC_MODE)" />
  
  <arg name="ROBOT" default="robot" />
  
  <!-- detect handle_pose from marker on crank -->
  <group if="$(arg LAUNCH_HANDLE_DETECTOR)" >
    <include file="$(find drc_task_common)/launch/fc/vehicle/handle_pose_detector.launch"/>
  </group>

  <!-- vehicle_ui -->
  <group ns="$(arg OCS_NS)" if="$(arg LAUNCH_UI)">
    <node pkg="drc_task_common" type="vehicle_ui" name="vehicle_ui" respawn="true" >
      <remap from="/chest_camera/image_color" to="/ocs/communication/panorama_image"/>
      <remap from="/multisense/left/image_rect_color" to="/ocs/communication/image_rect_color"/>
    </node>
  </group>

  <!-- handle controller -->
  <group if="$(arg LAUNCH_STEERING)">
    <include file="$(find drc_task_common)/launch/fc/vehicle/handle_operation_interface.launch">
      <arg name="CONTROLLER_DEV" default="$(arg CONTROLLER_DEV)"/>
      <arg name="OCS_NS" default="$(arg OCS_NS)"/>
    </include>
  </group>

  <!-- ps3joy controller -->
  <group if="$(arg LAUNCH_PS3JOY)">
    <include file="$(find jsk_teleop_joy)/launch/joy_vehicle.launch">
      <arg name="CONTROLLER_DEV" default="$(arg CONTROLLER_DEV)"/>
      <arg name="OCS_NS" default="$(arg OCS_NS)"/>
    </include>
  </group>
  
  <!-- eus -->
  <group if="$(arg LAUNCH_EUS)">
    <node pkg="roseus" type="roseus" name="drive_controller"
          args="$(find drc_task_common)/euslisp/vehicle/$(arg ROBOT)-driving-controller-on-polaris-xp900.l '($(arg ROBOT)-vehicle-main)'" />
  </group>

  <group if="$(arg LAUNCH_RVIZ)">
    <node pkg="rviz" type="rviz" name="vehicle_rviz" args="-d $(find drc_task_common)/config/vehicle_operator.rviz"
          respawn="true"/>
  </group>

  <!-- recognition -->
  <group if="$(arg LAUNCH_CAR_CENTER)">
    <include file="$(find drc_task_common)/launch/fc/vehicle/car_center_tf_publisher.launch">
      <arg name="REAL_MODE" default="true"/>
      <arg name="JAXON" default="true"/>
      <arg name="HRP2" default="false"/>
    </include>
  </group>
  <group if="$(arg LAUNCH_RECOGNITION)">
    <node name="drive_recognition" pkg="drc_task_common" type="drive_recognition.sh" args="real 0" />
  </group>
</launch>
